src/repl/scala/tools/nsc/interpreter/ILoop.scala

```diff
@@ -915,7 +916,8 @@ class ILoop(in0: Option[BufferedReader], protected val out: JPrintWriter)
       !intp.reporter.hasErrors
     }
     loadFiles(settings)
-    printWelcome()
+    if( ! Properties.propOrFalse( "scala.repl.disableWelcomeMessage" ) ) 
+      printWelcome()
```

Comment:
[Pasting my comment onto the diff.]

`ReplProps` collects various props that you can never remember the name of. Maybe time for `~/.scalarc`.

Would also be neat if the string property were a dollar quote, naturally with `$intp` in scope. If there are no dollars (a constant), spit it out immediately; otherwise, compile it and echo it in `loopPostInit`. An empty string means no banner. That's just an idea.

https://github.com/scala/scala/blob/2.11.x/src/repl/scala/tools/nsc/interpreter/ReplProps.scala

Also, the surrounding source is not quite so space happy.